### PR TITLE
Added Vector2 configurable

### PR DESCRIPTION
**Pull request by starrodkirby86**
So I totally wasn't sniping and peeking through your addition of the Color4 configurable to get an understanding of what was going on. :P

I wasn't sure on the naming, but I apologize in advance if there's some ultimate disorganization that you have to clean up on your end, should you merge it with open arms.

### CODE REVIEW

File: editor/UserInterface/Vector2Config.cs [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+            OnValueCommited?.Invoke(this, EventArgs.Empty);
+        }
+
+        private void updateWidgets()
+        {
+            xTextbox.SetValueSilent(value.X.ToString());
+            yTextbox.SetValueSilent(value.Y.ToString());
+        }
+
+        protected override void ApplyStyle(WidgetStyle style)
```
Comment (Damnae) [inline editor/UserInterface/Vector2Config.cs]: You don't need these 3 overrides

File: editor/UserInterface/Vector2Config.cs [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+            xTextbox.OnValueCommited += xTextbox_OnValueCommited;
+            yTextbox.OnValueCommited += yTextbox_OnValueCommited;
+        }
+
+        private void xTextbox_OnValueCommited(object sender, EventArgs e)
+        {
+            var xCommit = xTextbox.Value;
+
+            float X;
```
Comment (Damnae) [inline editor/UserInterface/Vector2Config.cs]: lowercase

File: editor/UserInterface/Vector2Config.cs [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+            }
+            Value = new Vector2(X , value.Y);
+            OnValueCommited?.Invoke(this, EventArgs.Empty);
+        }
+
+        private void yTextbox_OnValueCommited(object sender, EventArgs e)
+        {
+            var yCommit = yTextbox.Value;
+
+            float Y;
```
Comment (Damnae) [inline editor/UserInterface/Vector2Config.cs]: lowercase

File: editor/UserInterface/Vector2Config.cs [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+
+        protected override void Dispose(bool disposing)
+        {
+            base.Dispose(disposing);
+        }
+
+        protected override void Layout()
+        {
+            base.Layout();
+            layout.Size = new Vector2(Size.X, layout.PreferredSize.Y);
```
Comment (Damnae) [inline editor/UserInterface/Vector2Config.cs]: `layout.Size = Size;` is enough

File: editor/UserInterface/Vector2Config.cs (hunk 2 above)
Comment (starrodkirby86) [inline editor/UserInterface/Vector2Config.cs]: I wanted to keep in consistency with Vector2's convention of capitalizing the X and Y values for their components. But changed, nevertheless.

Comment (Damnae) [inline editor/UserInterface/Vector2Config.cs]: they're capitalize because they are public members of the struct

**Pull request merged**